pkg/user/signer_test.go

```diff
@@ -69,12 +71,44 @@ func (s *SignerTestSuite) TestSubmitTx() {
 	require.EqualValues(t, 0, resp.Code)
 }
 
-func (s *SignerTestSuite) ConfirmTxTimeout() {
-	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+// TestConfirmTx verifies that the ConfirmTx method returns the expected results.
+func (s *SignerTestSuite) TestConfirmTx() {
```

Comment:
[blocking] Can the test cases in this test be refactored to use table driven tests?

Ref: https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Reply:
Thanks for suggest! But I think this seems difficult to do, because each case requires different steps and inputs. For example, some cases need to call submit tx with different msg, some cases do not need to.

What do you think about dividing it into different function: TestConfirmTx_Timeout, TestConfirmTx_NotFoundTx,....?

Reply:
Hmm if the setup is distinct for each test case, we could use [this pattern](https://github.com/celestiaorg/celestia-app/blob/ebd733d1db2e75d849a637f68587534447ceccd0/pkg/square/square_test.go#L38-L53) with multiple `t.Run("test case for...")`

Reply:
oh thanks, I didn't know about this...

Reply:
@rootulp I just push a new commit. Please review it, thank you!

Reply:
Awesome, looks great!